Quarkus build with quarkus.container-image.builder=jib does not consider podman's auth.json

### Describe the bug

I get the exception below when i try to build my quarkus app after login in using `podman login` but without the properties `quarkus.container-image.username` and `quarkus.container-image.password`
```
[ERROR] Caused by: com.google.cloud.tools.jib.api.RegistryUnauthorizedException: Unauthorized for ...
```
Not sure if i am looking at the right place but when i look at the code [here](https://github.com/quarkusio/quarkus/blob/3.3.1/extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java#L356) it does not consider the podman authfile in `${XDG_RUNTIME_DIR}/containers/auth.json.`

as an alternative i was able to create a sym link - $HOME/.docker/config.json to the podman's auth file to get it working. 

It will be nice to add the podman's authfile if the docker config is not found.

### Expected behavior

The build should work and should produce a container image

### Actual behavior

The build throws an exception 
```
[ERROR] Caused by: com.google.cloud.tools.jib.api.RegistryUnauthorizedException: Unauthorized for ...
```

### How to Reproduce?

1. Remove `$home/.docker/config.json`
2. Before running the build, authenticate to the container registry using `podman login`
3. Run the build without the properties `quarkus.container-image.username` and `quarkus.container-image.password`

### Output of `java -version`
```
OpenJDK Runtime Environment GraalVM CE 17.0.8+7.1 (build 17.0.8+7-jvmci-23.0-b15) 
OpenJDK 64-Bit Server VM GraalVM CE 17.0.8+7.1 (build 17.0.8+7-jvmci-23.0-b15, mixed mode, sharing)
```
### Quarkus version or git rev

4.3.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.3 (21122926829f1ead511c958d89bd2f672198ae9f)

--- a/extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java
+++ b/extensions/container-image/container-image-jib/deployment/src/main/java/io/quarkus/container/image/jib/deployment/JibProcessor.java
@@ -362,6 +362,17 @@ public class JibProcessor {
         } else {
             registryImage.addCredentialRetriever(credentialRetrieverFactory.wellKnownCredentialHelpers());
             registryImage.addCredentialRetriever(credentialRetrieverFactory.dockerConfig());
+
+            // podman credentials: https://docs.podman.io/en/latest/markdown/podman-login.1.html
+            // podman for Windows and macOS
+            registryImage.addCredentialRetriever(credentialRetrieverFactory
+                    .dockerConfig(Paths.get(System.getProperty("user.home"), ".config", "containers", "auth.json")));
+            String xdgRuntimeDir = System.getenv("XDG_RUNTIME_DIR");
+            if ((xdgRuntimeDir != null) && !xdgRuntimeDir.isEmpty()) {
+                registryImage.addCredentialRetriever(
+                        credentialRetrieverFactory.dockerConfig(Paths.get(xdgRuntimeDir, "containers", "auth.json")));
+            }
+
             String dockerConfigEnv = System.getenv().get("DOCKER_CONFIG");
             if (dockerConfigEnv != null) {
                 Path dockerConfigPath = Path.of(dockerConfigEnv);

